client/my-sites/checkout/src/lib/razorpay-processor.ts

```diff
@@ -86,9 +85,13 @@ export default async function razorpayProcessor(
 				const razorpayOptions = combineRazorpayOptions(
 					transactionOptions.razorpayConfiguration,
 					response as WPCOMTransactionEndpointResponseRedirect,
-					( result ) => resolve( { bd_order_id: response.order_id.toString(), ...result } )
+					( result ) => {
+						transactionOptions.formStatusController.setFormSubmitting();
+						resolve( { bd_order_id: response.order_id.toString(), ...result } );
+					}
 				);
-				debug( 'Open Razorpay modal' );
+				debug( 'Opening Razorpay modal' );
+				transactionOptions.formStatusController.setFormSubmitting();
```

Comment:
🤔 Shouldn't the form status already be submitting at this point?

Reply:
It should, but for some reason it's not always. I can't consistently reproduce, but sometimes there is a brief period between the modal closing and the thank you page redirect when the checkout form is visible and ready.

Reply:
Ah! I think I know what that is. We've discovered a bug (https://github.com/Automattic/payments-shilling/issues/2379) where the form can return to "ready" status even while a transaction is still pending.

Reply:
Yep- that sounds exactly like what's happening here. I'll remove the extraneous form actions.